tests/unittests/classification/test_cohen_kappa.py

```diff
@@ -1,133 +1,337 @@
+    def test_multiclass_confusion_matrix_functional(self, input, weights, ignore_index):
+        preds, target = input
+        if ignore_index is not None:
+            target = inject_ignore_index(target, ignore_index)
         self.run_functional_metric_test(
-            preds,
-            target,
-            metric_functional=cohen_kappa,
-            sk_metric=partial(sk_metric, weights=weights),
-            metric_args={"num_classes": num_classes, "threshold": THRESHOLD, "weights": weights},
+            preds=preds,
+            target=target,
+            metric_functional=multiclass_cohen_kappa,
+            sk_metric=partial(_sk_cohen_kappa_multiclass, weights=weights, ignore_index=ignore_index),
+            metric_args={
+                "num_classes": NUM_CLASSES,
+                "weights": weights,
+                "ignore_index": ignore_index,
+            },
         )
 
-    def test_cohen_kappa_differentiability(self, preds, target, sk_metric, weights, num_classes):
+    def test_multiclass_cohen_kappa_differentiability(self, input):
+        preds, target = input
         self.run_differentiability_test(
             preds=preds,
             target=target,
-            metric_module=CohenKappa,
-            metric_functional=cohen_kappa,
-            metric_args={"num_classes": num_classes, "threshold": THRESHOLD, "weights": weights},
+            metric_module=MulticlassCohenKappa,
+            metric_functional=multiclass_cohen_kappa,
+            metric_args={"num_classes": NUM_CLASSES},
         )
 
+    @pytest.mark.parametrize("dtype", [torch.half, torch.double])
+    def test_multiclass_cohen_kappa_dtypes_cpu(self, input, dtype):
+        preds, target = input
+        if dtype == torch.half and not _TORCH_GREATER_EQUAL_1_6:
+            pytest.xfail(reason="half support of core ops not support before pytorch v1.6")
+        if (preds < 0).any() and dtype == torch.half:
+            pytest.xfail(reason="torch.sigmoid in metric does not support cpu + half precision")
+        self.run_precision_test_cpu(
+            preds=preds,
+            target=target,
+            metric_module=MulticlassCohenKappa,
+            metric_functional=multiclass_cohen_kappa,
+            metric_args={"num_classes": NUM_CLASSES},
+            dtype=dtype,
+        )
+
+    @pytest.mark.skipif(not torch.cuda.is_available(), reason="test requires cuda")
+    @pytest.mark.parametrize("dtype", [torch.half, torch.double])
+    def test_multiclass_confusion_matrix_dtypes_gpu(self, input, dtype):
+        preds, target = input
+        self.run_precision_test_gpu(
+            preds=preds,
+            target=target,
+            metric_module=MulticlassCohenKappa,
+            metric_functional=multiclass_cohen_kappa,
+            metric_args={"num_classes": NUM_CLASSES},
+            dtype=dtype,
+        )
+
+
+# -------------------------- Old stuff --------------------------
```

Comment:
shall we remove it compltely?

Reply:
I keep all the old stuff around for later, not sure if I am going to need it.
Will probably be removed in one of the last PRs.